consensus/misc/create2deployer.go

```diff
@@ -28,8 +28,9 @@ func init() {
 	create2DeployerCode = code
 }
 
-func EnsureCreate2Deployer(c *params.ChainConfig, timestamp uint64, db vm.StateDB) {
-	if !c.IsOptimism() || c.CanyonTime == nil || *c.CanyonTime != timestamp {
+func EnsureCreate2Deployer(c *params.ChainConfig, parentTimestamp, timestamp uint64, db vm.StateDB) {
+	// only ensure if Canyon is active, and if we just passed the fork-boundary or if this is genesis.
+	if !(c.IsOptimism() && c.IsCanyon(timestamp) && (!c.IsCanyon(parentTimestamp) || parentTimestamp == 0)) {
```

Comment:
Because https://github.com/ethereum-optimism/optimism/pull/8129, canyon enabled genesis state will contain create2deployer. Therefore, when block number `1` is building(`parentTimestamp == 0`) on optimism canyon enabled(`c.IsOptimism() && c.IsCanyon(timestamp)`) chain, It will still call `db.SetCode` although the desired bytecode is already there. This is not a problem, but I thought it was worth mentioning. We may remove `parentTimestamp == 0` condition.